apollo-runtime/src/main/java/com/apollographql/apollo/response/OperationResponseParser.java

```diff
@@ -49,8 +49,8 @@ public OperationResponseParser(Operation<D, W, ?> operation, ResponseFieldMapper
     checkNotNull(payload, "payload == null");
 
     D data = null;
-    if (payload.containsKey("data")) {
-      Map<String, Object> buffer = (Map<String, Object>) payload.get("data");
+    Map<String, Object> buffer = (Map<String, Object>) payload.get("data");
+    if (buffer!=null) {
```

Comment:
Am I missing smth but the changes are identical to what was before?

Reply:
Sorry my bad you are totally right.
Could you please add some test as well?

Reply:
Added simple passing unit test.